### ISSUE TITLE
[Bug] Tries to create existing amin account

### ISSUE THREAD

**Issue by henri9813** [AppFlowy-IO/AppFlowy-Cloud#544]
**Describe the bug**
At each appflowy cloud start

```
appflowy_cloud-1  | {"timestamp":"2024-05-09T11:39:38.105850Z","level":"ERROR","fields":{"error":"code: 400, msg:User already registered, error_id: None"},"target":"gotrue::api","span":{"name":"sign_up"},"spans":[{"name":"sign_up"}]}
```

When we configure gotrue with:
```
      - GOTRUE_DISABLE_SIGNUP=true
```

Appflowy is not able to start at all
```
appflowy_cloud-1  | {"timestamp":"2024-05-09T11:38:20.383807Z","level":"ERROR","fields":{"error":"code: 403, msg:Signups not allowed for this instance, error_id: None"},"target":"gotrue::api","span":{"name":"sign_up"},"spans":[{"name":"sign_up"}]}
appflowy_cloud-1  | Error: Failed to initialize application state: code: 403, msg:Signups not allowed for this instance, error_id: None
appflowy_cloud-1  | 
appflowy_cloud-1  | Stack backtrace:
appflowy_cloud-1  |    0: anyhow::error::<impl anyhow::Error>::msg
appflowy_cloud-1  |    1: tokio::task::local::LocalSet::run_until::{{closure}}
appflowy_cloud-1  |    2: appflowy_cloud::main
appflowy_cloud-1  |    3: std::sys_common::backtrace::__rust_begin_short_backtrace
appflowy_cloud-1  |    4: main
appflowy_cloud-1  |    5: <unknown>
appflowy_cloud-1  |    6: __libc_start_main
appflowy_cloud-1  |    7: _start
```

**To Reproduce**
1. Start appflowy docker-compose.

**Expected behavior**
User already exist, no error triggered.

**Desktop (please complete the following information):**

**Additional context**
 - Environment [e.g. flutter doctor -v or rustup show]

**Comment by ppluciennik** [comment]
This patch appears to fix the startup issue:

```
diff --git a/src/application.rs b/src/application.rs
index 69e093c..0bc9470 100644
--- a/src/application.rs
+++ b/src/application.rs
@@ -320,6 +320,7 @@ async fn setup_admin_account(
     Err(err) => {
       if let app_error::gotrue::GoTrueError::Internal(err) = err {
         match (err.code, err.msg.as_str()) {
+          (403, "Signups not allowed for this instance") => Ok(gotrue_admin),
           (400, "User already registered") => {
             info!("Admin user already registered");
             Ok(gotrue_admin)
```

**Comment by henri9813** [comment]
Hello,

Should we not just check if the admin account exists before trying to create it and wait an error ?

**Comment by ppluciennik** [comment]
That's better option.

**Comment by Killian-Aidalinfo** [comment]
Hello, I have the same problem, is there a way around it?

**Comment by henri9813** [comment]
I will learn rust and then propose a patch if no one do this :)

**Comment by khorshuheng** [comment]
Currently, these are the steps that we expect if a user does not want to allow sign ups:
https://github.com/AppFlowy-IO/AppFlowy-Cloud/blob/main/doc/DEPLOYMENT.md#how-do-i-disable-signups

At the moment, AppFlowy Cloud service is the one that is responsible for creating gotrue admin user. Therefore, if sign up is disabled, this admin user cannot be created. Once the admin user is created, the sign up can be disabled without issue.

It is certainly inconvenient, and could have been automated by having one time service/job/script that create the gotrue admin user before appflowy cloud starts. Say, using the gotrue CLI admin command. However, this does make the docker compose setup more complicated, and since the steps described in the deployment doc only need to be performed once, we decided not to automate this for now.

**Comment by henri9813** [comment]
Hello,

I do not agree at all with the closing of the issue.

Why AppFlowy cannot verify the user exists before trying to create a new one ?

Furthermore: "we decide to not fix it now" doesn't mean the subject doesn't exist. So the issue must remain open.

**Comment by khorshuheng** [comment]
Fixed in release 0.8.4: https://github.com/AppFlowy-IO/AppFlowy-Cloud/releases/tag/0.8.4

New users should be able to deploy appflowy cloud with GOTRUE_DISABLE_SIGNUP=false without issue, as long as the .env variable is based on the updated deploy.env file, and the docker compose file is also updated.

**Comment by henri9813** [comment]
Thank you @khorshuheng !!!
